test/schema_test.exs

```diff
@@ -62,4 +62,17 @@ defmodule OpenApiSpex.SchemaTest do
     assert_schema(Schemas.UserResponse.schema().example, "UserResponse", spec)
     assert_schema(Schemas.UsersResponse.schema().example, "UsersResponse", spec)
   end
-end
\ No newline at end of file
+
+  test "Cast Cat from Pet schema" do
+    api_spec = ApiSpec.spec()
+    schemas = api_spec.components.schemas
+    pet_schema = schemas["Pet"]
+
+    input = %{
```

Comment:
This map should have string keys.

Reply:
I've just addressed this in 3fe61f94aa0d91800c4e6a8512ce98be5c40f50c